### PR TITLE
CDAP-15100 Improve performance of DBRecord class

### PR DESCRIPTION
**Pull request by mlozbin-cybervisiontech**
JIRA: https://issues.cask.co/browse/CDAP-15100
WIKI: https://wiki.cask.co/display/CE/Database+plugin+enhancements

In scope of this PR:
- Removed input schema parsing on each incoming row
- Removed field validation on each incoming row
- Added schema validation on pipeline deployment
- Schema parsing changed to lazy computation with cache
- Actual DB schema resolving was moved to "prepareRun" stage

Attachments ([CDAP-15100_report.zip](https://github.com/data-integrations/database-plugins/files/3088404/CDAP-15100_report.zip)):
- CPU utilization flamegraphs before and after improvement
- CDAP report for 500k records read before and after improvement for same pipeline
- Test pipeline

### CODE REVIEW

**Comment by mlozbin-cybervisiontech** [comment]
@jayfjin could you, please, take a look

File: database-commons/src/main/java/io/cdap/plugin/util/Lazy.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public final class Lazy<T> {
+  private final Supplier<T> supplier;
+  private volatile T value;
+
+  public Lazy(Supplier<T> supplier) {
+    this.supplier = supplier;
+  }
+
+  public T getOrCompute() {
+    final T result = value; // Just one volatile read
```
Comment (jayfjin) [inline database-commons/src/main/java/io/cdap/plugin/util/Lazy.java:37]: does this function need to be atomic as well?

**Review by jayfjin — COMMENTED** [review]
Great work! Just one question

File: database-commons/src/main/java/io/cdap/plugin/util/Lazy.java (hunk 1 above)
Comment (mlozbin-cybervisiontech) [reply database-commons/src/main/java/io/cdap/plugin/util/Lazy.java:37]: do you mean synchronized? If so, then answer is no. This helper uses Double-checked locking in its best implementation.

**Pull request merged**